tests/components/screenlogic/test_init.py

```diff
@@ -0,0 +1,166 @@
+        "pump_0_watts_now",
+        SENSOR_DOMAIN,
+    ),
+    EntityMigrationData(
+        "SCG Status",
+        "scg_status",
+        "Chlorinator",
+        "scg_state",
+        BINARY_SENSOR_DOMAIN,
+    ),
+    EntityMigrationData(
+        "Non-Migrating Sensor",
+        "nonmigrating",
+        "Non-Migrating Sensor",
+        "nonmigrating",
+        SENSOR_DOMAIN,
+    ),
+    EntityMigrationData(
+        "Missing Migration Device",
+        "missing_device",
+        "Missing Migration Device",
+        "missing_device",
+        BINARY_SENSOR_DOMAIN,
+    ),
+    EntityMigrationData(
+        "Old Sensor",
+        "old_sensor",
+        "Old Sensor",
+        "old_sensor",
+        SENSOR_DOMAIN,
+    ),
+]
+
+
+@pytest.mark.parametrize(
+    ("entity_def", "ent_data"),
+    [
+        (
+            {
+                "domain": ent_data.domain,
+                "platform": DOMAIN,
+                "unique_id": f"{MOCK_ADAPTER_MAC}_{ent_data.old_key}",
+                "suggested_object_id": f"{MOCK_ADAPTER_NAME} {ent_data.old_name}",
+                "disabled_by": None,
+                "has_entity_name": True,
+                "original_name": ent_data.old_name,
+            },
+            ent_data,
+        )
+        for ent_data in TEST_MIGRATING_ENTITIES
+    ],
+)
+async def test_async_migrate_entries(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    mock_migration_gateway,
+    entity_def: dict,
+    ent_data: EntityMigrationData,
+) -> None:
+    """Test migration to new entity names."""
+
+    mock_config_entry.add_to_hass(hass)
+
+    entity_registry = er.async_get(hass)
+    device_registry = dr.async_get(hass)
+
+    device: dr.DeviceEntry = device_registry.async_get_or_create(
+        config_entry_id=mock_config_entry.entry_id,
+        identifiers={(DOMAIN, MOCK_ADAPTER_MAC)},
+    )
+
+    TEST_EXISTING_ENTRY = {
+        "domain": SENSOR_DOMAIN,
+        "platform": DOMAIN,
+        "unique_id": f"{MOCK_ADAPTER_MAC}_existing",
+        "suggested_object_id": f"{MOCK_ADAPTER_NAME} Existing Sensor",
+        "disabled_by": None,
+        "has_entity_name": True,
+        "original_name": "Existing Sensor",
+    }
+
+    entity_registry.async_get_or_create(
+        **TEST_EXISTING_ENTRY, device_id=device.id, config_entry=mock_config_entry
+    )
+
+    entity: er.RegistryEntry = entity_registry.async_get_or_create(
+        **entity_def, device_id=device.id, config_entry=mock_config_entry
+    )
+
+    old_eid = f"{ent_data.domain}.{slugify(f'{MOCK_ADAPTER_NAME} {ent_data.old_name}')}"
+    old_uid = f"{MOCK_ADAPTER_MAC}_{ent_data.old_key}"
+    new_eid = f"{ent_data.domain}.{slugify(f'{MOCK_ADAPTER_NAME} {ent_data.new_name}')}"
+    new_uid = f"{MOCK_ADAPTER_MAC}_{ent_data.new_key}"
+
+    assert entity.unique_id == old_uid
+    assert entity.entity_id == old_eid
+    with patch.dict(
+        "homeassistant.components.screenlogic.data.ENTITY_MIGRATIONS",
```

Comment:
Can this be done without patching the integration itself?

https://developers.home-assistant.io/docs/development_testing?_highlight=writing&_highlight=te#writing-tests-for-integrations

Reply:
I'd split it and add a comment about why its behind patched

Reply:
[c7b6b34](https://github.com/home-assistant/core/pull/92475/commits/c7b6b346b87ffca20700b2c257a710a6f9d7963c)